src/QueryPipeline/QueryPipeline.cpp

```diff
@@ -529,7 +530,7 @@ void QueryPipeline::addCompletedPipeline(QueryPipeline other)
         throw Exception(ErrorCodes::LOGICAL_ERROR, "Cannot add not completed pipeline");
 
     resources = std::move(other.resources);
-    processors.insert(processors.end(), other.processors.begin(), other.processors.end());
+    processors->insert(processors->end(), other.processors->begin(), other.processors->end());
 }
 
 void QueryPipeline::reset()
```

Comment:
Here it will simply reset the shared_ptr instead of modifying the shared std::vector.